### PR TITLE
fix: disable AccountForm when user is not allowed edit users

### LINKED ISSUE #3286: Member user cannot update their username
## Problem statement

When a member user tries to update their username, they are thrown an error as seen in the following screenshot:
Screen Shot 2022-07-28 at 11 26 50 PM

## Proposal

Either we should allow members to update their own username or have a better message that can provide them the steps to do it (e.g. requesting their admin), or clarify that they don't have access to do it.

## Definition of done

TBD

### PR DESCRIPTION
**Pull request by johnstcn**
Fixes https://github.com/coder/coder/issues/3286

### CODE REVIEW

**Review by Emyrk — APPROVED** [review]
Just need to disable the update settings button :+1:

File: site/src/pages/UserSettingsPage/AccountPage/AccountPage.tsx [hunk 1]
```diff
@@ -20,10 +21,13 @@ export const AccountPage: React.FC = () => {
   return (
     <Section title={Language.title}>
       <AccountForm
+        editable={!!canEditUsers}
```
Comment (jsjoeio) [inline site/src/pages/UserSettingsPage/AccountPage/AccountPage.tsx:24]: nit: `Boolean()` feels more readable than `!!`. Probably something that should be discussed and if agreed by team, enforced via ESLint [`no-implicit-coercion`](https://eslint.org/docs/latest/rules/no-implicit-coercion)
```suggestion
        editable={Boolean(canEditUsers)}
```

**Review by jsjoeio — APPROVED** [review]
Looking fabulous!

File: site/src/pages/UserSettingsPage/AccountPage/AccountPage.tsx (hunk 1 above)
Comment (johnstcn) [reply site/src/pages/UserSettingsPage/AccountPage/AccountPage.tsx:24]: Yeah I just got impatient here 😅

Comment (jsjoeio) [reply site/src/pages/UserSettingsPage/AccountPage/AccountPage.tsx]: Added to next FE Variety to discuss!

Comment (jsjoeio) [reply site/src/pages/UserSettingsPage/AccountPage/AccountPage.tsx]: Lol can't blame you!

**Pull request merged**